tests/test_fitsChan.py

```diff
@@ -13,6 +13,58 @@ def pad(card):
     return "%-80s" % (card,)
 
 
+def writeFitsWcs(frameSet, extraOptions=None):
+    """Write a FrameSet as FITS-WCS
+
+    extraOptions are in addition to Encoding=Fits-WCS, CDMatrix=1
+    """
+    options = "Encoding=FITS-WCS, CDMatrix=1"
+    if extraOptions is not None:
+        options = "%s, %s" % (options, extraOptions)
+    fc = ast.FitsChan(ast.StringStream(), options)
+    fc.write(frameSet)
+    return fc
+
+
+def insertPixelMapping(mapping, frameSet):
+    """Make a new WCS by inserting a new mapping at the beginnning of the GRID-IWC mapping
+
+    Return the new FrameSet (the original is not altered).
+    """
+    frameSet = frameSet.copy()
+
+    skyFrame = frameSet.getFrame(ast.FrameSet.CURRENT)  # use this copy for the new sky frame
+    assert isinstance(skyFrame, ast.SkyFrame)
+    oldSkyIndex = frameSet.current
+
+    if not frameSet.findFrame(ast.Frame(2, "Domain=GRID")):
+        raise KeyError("No GRID frame")
+    gridIndex = frameSet.current
+
+    if not frameSet.findFrame(ast.Frame(2, "Domain=IWC")):
+        raise KeyError("No IWC frame")
+    oldIwcIndex = frameSet.current
+    iwcFrame = frameSet.getFrame(oldIwcIndex)  # use this copy for the new IWC frame
+
+    oldGridToIwc = frameSet.getMapping(gridIndex, oldIwcIndex)
+    iwcToSky = frameSet.getMapping(oldIwcIndex, oldSkyIndex)
+
+    # Remove frames in order high to low, so removal doesn't alter the indices remaining to be removed
+    # update gridIndex at the same time. Updating pixel0Index is paranoia, in case we
+    # find a use for it later. The current code is done with pixel0Index by now: the BASE frame remains
+    # unchanged throughout the code, so it does not have to be reset at the end.
+    framesToRemove = reversed(sorted([oldIwcIndex, oldSkyIndex]))
+    for index in framesToRemove:
+        if (index < gridIndex):
```

Comment:
This is shifting the current frame position down each time a frame with a lower index is removed?

Reply:
Yes, exactly.